src/libs/ajax/data-table-providers/WdsDataTableProvider.ts

```diff
@@ -82,46 +83,70 @@ const getRelationParts = (val: unknown): string[] => {
   return []
 }
 
-//Although this method repeats some logic from the resolveWdsUrl method, the two are using somewhat different criteria
-export const resolveWdsApp = apps => {
-  const namedApp = apps.filter(app => app.appType === 'CROMWELL' && app.appName === `wds-${app.workspaceId}` && ['RUNNING', 'PROVISIONING', 'STOPPED', 'STOPPING'].includes(app.status))
+export const createLeoAppWithErrorHandling = workspaceId => {
+  const typedWithErrorReporting : any = withErrorReporting
+  const createLeoAppCall = typedWithErrorReporting('An error occurred when creating your data tables. Please reach out to support@terra.bio', async () => {
+    await Ajax().Apps.createAppV2(`wds-${workspaceId}`, `${workspaceId}`)
+  })
+  createLeoAppCall()
+}
+
+// Invokes logic to determine the appropriate app for WDS
+// If WDS is not running, a URL will not be present -- in some cases, this function may invoke
+// a new call to Leo to instantiate a WDS being available, thus having a valid URL
+export const resolveWdsApp = (apps, workspaceId, shouldAutoDeployWds) => {
+  // WDS looks for Kubernetes deployment statuses (such as RUNNING or PROVISIONING), expressed by Leo
+  // See here for specific enumerations -- https://github.com/DataBiosphere/leonardo/blob/develop/core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala
+  // look explicitly for a RUNNING app named 'wds-${app.workspaceId}' -- if WDS is healthy and running, there should only be one app RUNNING
+  // an app may be in the 'PROVISIONING', 'STOPPED', 'STOPPING', which can still be deemed as an OK state for WDS
+  const healthyStates = ['RUNNING', 'PROVISIONING', 'STOPPED', 'STOPPING']
+  const namedApp = apps.filter(app => app.appType === 'CROMWELL' && app.appName === `wds-${app.workspaceId}` && healthyStates.includes(app.status))
   if (namedApp.length === 1) {
     return namedApp[0]
   }
 
-  //Failed to find an app with the proper name, look for a RUNNING CROMWELL app
+  // if we didn't find the expected app 'wds-${app.workspaceId}' running...
+  const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.appName === `wds-${app.workspaceId}`)
+  // ...nothing has launched yet, bring WDS to life if the request is coming from the OWNER of the workspace!
```

Comment:
I know we've discussed this logic ad nauseum, but what if there is a wds running with an unusual name?  Would we still want to launch?

Reply:
I think I addressed this in https://github.com/DataBiosphere/terra-ui/pull/3753